localstack/services/apigateway/provider.py

```diff
@@ -1751,7 +1759,15 @@ def get_integration(
     def put_integration(
         self, context: RequestContext, request: PutIntegrationRequest
     ) -> Integration:
-        if request.get("type") == IntegrationType.AWS_PROXY:
+        if (integration_type := request.get("type")) not in VALID_INTEGRATION_TYPES:
```

Comment:
interesting...we don't have many negative tests around this - I wonder if some aws sdks validate this before it actually makes the request.

Reply:
Yes, for example boto validates and does not let you send the request. But I suppose some SDK are more lenient. We've often relied on SDK validation to do the work for us, but now can see we even need to disable boto validation to be able to test real AWS behavior...